src/mount-support.c

```diff
@@ -215,6 +215,22 @@ void setup_snappy_os_mounts()
 			die("cannot bind mount %s to %s", src, dst);
 		}
 	}
+	// Don't expose /etc/alternatives from the host, ensure that
+	// /etc/alternatives is always the version from the core snap.
+	// https://bugs.launchpad.net/snap-confine/+bug/1580018
+	const char *etc_alternatives = "/etc/alternatives";
+	if (access(etc_alternatives, F_OK) == 0) {
+		char src[512];
+		char dst[512];
+		must_snprintf(src, sizeof src, "%s%s", core_snap_dir,
+			      etc_alternatives);
+		must_snprintf(dst, sizeof dst, "%s%s", rootfs_dir,
+			      etc_alternatives);
+		debug("bind mounting %s to %s", src, dst);
+		if (mount(src, dst, NULL, MS_BIND | MS_SLAVE, NULL) != 0) {
```

Comment:
LGTM but please add a note about MS_SLAVE here like we do for the source_mounts mount() call above so it is clear.